### PR TITLE
Add local dev server

### PR DESCRIPTION
**Pull request by kazzarin**
Adds [serve](https://github.com/vercel/serve) as an optional package to locally host the `build` directory to allow the userscript to auto-update on script changes.

See the following for how this works: https://violentmonkey.github.io/posts/how-to-edit-scripts-with-your-favorite-editor/#install-a-local-script

Other changes:
- removed `ecmaVersion` from eslint config as the `es2018` key will [set it automatically](https://eslint.org/docs/latest/use/configure/language-options#specifying-environments)
- changed script extension to `.user.js` 
  - this is needed for the updater to work
- add build scripts
  - a shortcut for make command
  - also added one that uses [WSL](https://learn.microsoft.com/en-us/windows/wsl/about)
    - (I plan to add a guide for this though I wanted to rework the existing docs first)

### CODE REVIEW

**Review by alsoGAMER — APPROVED** [review]

**Comment by hohMiyazawa** [comment]
Will this in any way impact exiting ways of building the script?

**Comment by alsoGAMER** [comment]
For what i saw, i don't think so

File: .eslintrc.json [hunk 1]
```diff
@@ -2,33 +2,30 @@
     "root": true,
     "env": {
         "browser": true,
-        "es2017": true,
+        "es2018": true,
```
Comment (hohMiyazawa) [inline .eslintrc.json:5]: Why is this bumped?

File: src/makefile [hunk 2]
```diff
@@ -3,14 +3,14 @@ EXECUTABLES = sed rm mkdir date zip m4 cp cat
...
-all: pre-build build build/automail.js build/boneless.js build/firefox_addon.zip post-build
+all: pre-build build build/automail.user.js build/boneless.js build/firefox_addon.zip post-build
 
 pre-build:
 	rm -f build/userModules.js
 
-build/automail.js: automail.m4 settings.js alias.js css/global.css conditionalStyles.js utilities.js purify.js graphql.js localisation.js controller.js build/userModules.js HOWTO.js
-	m4 --prefix-builtins automail.m4 > build/automail.js
-	date +"%s" | sed 's_^_//Automail built at _' >> build/automail.js
+build/automail.user.js: automail.m4 settings.js alias.js css/global.css conditionalStyles.js utilities.js purify.js graphql.js localisation.js controller.js build/userModules.js HOWTO.js
```
Comment (alsoGAMER) [inline src/makefile:11]: the only thing you wanna check on greasy fork, just in case it breaks something is this
@hohMiyazawa

Comment (hohMiyazawa) [reply src/makefile:11]: Tested. The renaming does not affect the userscript upload procedure.

File: .eslintrc.json (hunk 1 above)
Comment (kazzarin) [reply .eslintrc.json:5]: At the time the config was introduced, it only had options for es versions that introduced new globals. It seems this changed in an eslint update from last year: https://eslint.org/blog/2022/02/eslint-v8.9.0-released/#highlights

>Previously, we used to add new ES environments only if they contain new globals compared to previous years. New es2016, es2018, and es2019 environments have also been added to fill in the gaps between years.

So this just aligns it to what was initially intended: ES9 (2018)

**Comment by kazzarin** [comment]
>Will this in any way impact exiting ways of building the script?

It shouldn't besides the file extension change

**Pull request merged**